homeassistant/components/whirlpool/sensor.py

```diff
@@ -0,0 +1,287 @@
+"""The Washer/Dryer Sensor for Whirlpool Appliances."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+
+from whirlpool.washerdryer import MachineState, WasherDryer
+
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.dt import utcnow
+
+from . import WhirlpoolData
+from .const import DOMAIN
+
+TANK_FILL = {
+    "0": "Unknown",
```

Comment:
All of these sensors should be enum device class sensors by the looks of the code.

The state string returned should then be lowercase snake_case and we should use a `translation_key`.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

https://developers.home-assistant.io/docs/internationalization/core?_highlight=translation_key#state-of-entities

Reply:
Device class is a bit tricky here. (Maybe I'm missing/misunderstanding your point)  For the tank fill, it's reported in strange values 0-5.  
I originally had it as volume, but after reviewing it, the value % doesn't fit volume.
Machine state doesn't appear to match any device classes either.

Reply:
When setting the enum device class, the sensor should report one of a specific set of strings. In this case it would be the right side of the `TANK_FILL` dict, after we change them to lowercase snake_case. We add translation strings in `strings.json` and set a `translation_key` attribute on the entity to allow more user friendly names of the states.

Reply:
Sorry, I missed the `enum` part. - works like a champ